src/typed_array.rs

```diff
@@ -0,0 +1,360 @@
+    /// let array = evaluate_script(
+    ///     &ctx,
+    ///     "new Uint8Array([1, 2, 3, 4, 5])",
+    ///     None,
+    ///     "foo.js",
+    ///     1,
+    /// )
+    ///     .unwrap()
+    ///     .as_typed_array()
+    ///     .unwrap();
+    /// assert_eq!(array.len().unwrap(), 5);
+    /// ```
+    #[allow(clippy::len_without_is_empty)]
+    pub fn len(&self) -> Result<usize, JSException> {
+        let mut exception: sys::JSValueRef = ptr::null_mut();
+        let value = unsafe { sys::JSObjectGetTypedArrayLength(self.ctx, self.raw, &mut exception) };
+
+        if !exception.is_null() {
+            Err(unsafe { JSValue::from_raw(self.ctx, exception) }.into())
+        } else {
+            Ok(value)
+        }
+    }
+
+    /// Returns the byte offset of the Typed Array.
+    ///
+    /// ```rust
+    /// # use javascriptcore::*;
+    /// let ctx = JSContext::default();
+    /// let array = evaluate_script(
+    ///     &ctx,
+    ///     "const array = new Uint8Array([1, 2, 3, 4, 5]); new Uint8Array(array.buffer, 3)",
+    ///     None,
+    ///     "foo.js",
+    ///     1,
+    /// )
+    ///     .unwrap()
+    ///     .as_typed_array()
+    ///     .unwrap();
+    /// assert_eq!(array.byte_offset().unwrap(), 3);
+    /// ```
+    pub fn byte_offset(&self) -> Result<usize, JSException> {
+        let mut exception: sys::JSValueRef = ptr::null_mut();
+        let offset =
+            unsafe { sys::JSObjectGetTypedArrayByteOffset(self.ctx, self.raw, &mut exception) };
+
+        if !exception.is_null() {
+            Err(unsafe { JSValue::from_raw(self.ctx, exception) }.into())
+        } else {
+            Ok(offset)
+        }
+    }
+
+    /// Returns the byte length of the Typed Array.
+    /// ```rust
+    /// # use javascriptcore::*;
+    /// let ctx = JSContext::default();
+    /// let array = evaluate_script(
+    ///     &ctx,
+    ///     "const array = new Uint8Array([1, 2, 3, 4, 5]); new Uint8Array(array.buffer, 1, 2)",
+    ///     None,
+    ///     "foo.js",
+    ///     1,
+    /// )
+    ///     .unwrap()
+    ///     .as_typed_array()
+    ///     .unwrap();
+    /// assert_eq!(array.byte_length().unwrap(), 2);
+    /// ```
+    pub fn byte_length(&self) -> Result<usize, JSException> {
+        let mut exception: sys::JSValueRef = ptr::null_mut();
+        let length =
+            unsafe { sys::JSObjectGetTypedArrayByteLength(self.ctx, self.raw, &mut exception) };
+
+        if !exception.is_null() {
+            Err(unsafe { JSValue::from_raw(self.ctx, exception) }.into())
+        } else {
+            Ok(length)
+        }
+    }
+
+    /// Returns a mutable slice of the underlying buffer represented by the
+    /// Typed Array.
+    ///
+    /// # Safety
+    ///
+    /// The pointer of the slice returned by this function is temporary and is not
+    /// guaranteed to remain valid across JavaScriptCore API calls.
+    pub unsafe fn as_mut_slice(&mut self) -> Result<&mut [u8], JSException> {
```

Comment:
A code example here would be good, also to make sure the borrowing part works out…

Reply:
Or maybe the tests are enough… could go either way. Maybe a minimal single example.